tests/protocol/test_hostplugin.py

```diff
@@ -365,15 +399,159 @@ def test_validate_get_extension_artifacts(self):
+
+        self.assertEqual(2, patch_http_post.call_count)
+
+        # signal for /versions
+        self.assertEqual(health_service_url, patch_http_post.call_args_list[0][0][0])
+        jstr = patch_http_post.call_args_list[0][0][1]
+        obj = json.loads(jstr)
+        self.assertEqual(1, len(obj['Observations']))
+        self.assertTrue(obj['Observations'][0]['IsHealthy'])
+        self.assertEqual('GuestAgentPluginVersions', obj['Observations'][0]['ObservationName'])
+
+        # signal for /status
+        self.assertEqual(health_service_url, patch_http_post.call_args_list[1][0][0])
+        jstr = patch_http_post.call_args_list[1][0][1]
+        obj = json.loads(jstr)
+        self.assertEqual(1, len(obj['Observations']))
+        self.assertTrue(obj['Observations'][0]['IsHealthy'])
+        self.assertEqual('GuestAgentPluginStatus', obj['Observations'][0]['ObservationName'])
+
+    @patch("azurelinuxagent.common.utils.restutil.http_get")
+    @patch("azurelinuxagent.common.utils.restutil.http_post")
+    @patch("azurelinuxagent.common.utils.restutil.http_put")
+    def test_put_status_unhealthy_signal_permanent(self, patch_http_put, patch_http_post, patch_http_get):
+        host_plugin = self._init_host()
+        status_blob = self._init_status_blob()
+        # get_api_versions
+        patch_http_get.return_value = MockResponse(api_versions, 200)
+        # put status blob
+        patch_http_put.return_value = MockResponse(None, 500)
+
+        host_plugin.status_error_state.is_triggered = Mock(return_value=True)
+
+        if sys.version_info < (2, 7):
+            self.assertRaises(HttpError, host_plugin.put_vm_status, status_blob, sas_url)
+        else:
+            with self.assertRaises(HttpError):
+                host_plugin.put_vm_status(status_blob=status_blob, sas_url=sas_url)
+
+        self.assertEqual(1, patch_http_get.call_count)
+        self.assertEqual(hostplugin_versions_url, patch_http_get.call_args[0][0])
+
+        self.assertEqual(1, patch_http_put.call_count)
+        self.assertEqual(hostplugin_status_url, patch_http_put.call_args[0][0])
+
+        self.assertEqual(2, patch_http_post.call_count)
+
+        # signal for /versions
+        self.assertEqual(health_service_url, patch_http_post.call_args_list[0][0][0])
+        jstr = patch_http_post.call_args_list[0][0][1]
+        obj = json.loads(jstr)
+        self.assertEqual(1, len(obj['Observations']))
+        self.assertTrue(obj['Observations'][0]['IsHealthy'])
+        self.assertEqual('GuestAgentPluginVersions', obj['Observations'][0]['ObservationName'])
+
+        # signal for /status
+        self.assertEqual(health_service_url, patch_http_post.call_args_list[1][0][0])
+        jstr = patch_http_post.call_args_list[1][0][1]
+        obj = json.loads(jstr)
+        self.assertEqual(1, len(obj['Observations']))
+        self.assertFalse(obj['Observations'][0]['IsHealthy'])
```

Comment:
> self.assertFalse(obj['Observations'][0]['IsHealthy']) [](start = 7, length = 54)

I think this is really what you want the test to assert, but it takes a lot of code to actually get there.  I have found this to be true for other parts of the agent. :cry: